server/src/main/java/org/opensearch/cluster/decommission/DecommissionController.java

```diff
@@ -0,0 +1,231 @@
+                    listener.onResponse(null);
+                }
+
+                @Override
+                public void handleException(TransportException exp) {
+                    listener.onFailure(exp);
+                }
+
+                @Override
+                public String executor() {
+                    return ThreadPool.Names.SAME;
+                }
+
+                @Override
+                public AddVotingConfigExclusionsResponse read(StreamInput in) throws IOException {
+                    return new AddVotingConfigExclusionsResponse(in);
+                }
+            }
+        );
+    }
+
+    public void clearVotingConfigExclusion(ActionListener<Void> listener) {
+        final ClearVotingConfigExclusionsRequest clearVotingConfigExclusionsRequest = new ClearVotingConfigExclusionsRequest();
+        clearVotingConfigExclusionsRequest.setWaitForRemoval(true);
+        transportService.sendRequest(
+            transportService.getLocalNode(),
+            ClearVotingConfigExclusionsAction.NAME,
+            clearVotingConfigExclusionsRequest,
+            new TransportResponseHandler<ClearVotingConfigExclusionsResponse>() {
+                @Override
+                public void handleResponse(ClearVotingConfigExclusionsResponse response) {
+                    logger.info("successfully cleared voting config after decommissioning");
+                    listener.onResponse(null);
+                }
+
+                @Override
+                public void handleException(TransportException exp) {
+                    logger.debug(new ParameterizedMessage("failure in clearing voting config exclusion after decommissioning"), exp);
+                    listener.onFailure(exp);
+                }
+
+                @Override
+                public String executor() {
+                    return ThreadPool.Names.SAME;
+                }
+
+                @Override
+                public ClearVotingConfigExclusionsResponse read(StreamInput in) throws IOException {
+                    return new ClearVotingConfigExclusionsResponse(in);
+                }
+            }
+        );
+    }
+
+    public void handleNodesDecommissionRequest(
+        Set<DiscoveryNode> nodesToBeDecommissioned,
+        String reason,
+        TimeValue timeout,
+        ActionListener<Void> nodesRemovedListener
+    ) {
+        final Map<NodeRemovalClusterStateTaskExecutor.Task, ClusterStateTaskListener> nodesDecommissionTasks = new LinkedHashMap<>();
+        nodesToBeDecommissioned.forEach(discoveryNode -> {
+            final NodeRemovalClusterStateTaskExecutor.Task task = new NodeRemovalClusterStateTaskExecutor.Task(discoveryNode, reason);
+            nodesDecommissionTasks.put(task, nodeRemovalExecutor);
+        });
+        clusterService.submitStateUpdateTasks(
+            "node-decommissioned",
+            nodesDecommissionTasks,
+            ClusterStateTaskConfig.build(Priority.IMMEDIATE),
```

Comment:
Should this be `URGENT` to let `IMMEDIATE` complete?

Reply:
Yes. Changing in next commit